### PR TITLE
HIVE-25372: Advance write id for all the DDLs

### PR DESCRIPTION
**Pull request by kishendas**
### What changes were proposed in this pull request?
Advance write ID during 12 DDLs

### Why are the changes needed?
HMS cache relies on ValidWriteIdList to serve the data consistently. 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Unit tests

### CODE REVIEW

**Comment by kishendas** [comment]
Got a green build -> http://ci.hive.apache.org/blue/organizations/jenkins/hive-precommit/detail/PR-2524/3/tests/

File: ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands.java [hunk 1]
```diff
@@ -432,6 +433,50 @@ public void testAddAndDropConstraintAdvancingWriteIds() throws Exception {
...
+
+    runStatementOnDriver(String.format("ALTER TABLE %s SET SERDEPROPERTIES ('field.delim'='\\u0001')", tableName));
+    validWriteIds  = msClient.getValidWriteIds("default." + tableName).toString();
+    Assert.assertEquals("default.alter_table:5:9223372036854775807::", validWriteIds);
+
+    runStatementOnDriver(String.format("ALTER TABLE %s PARTITION (ds='2013-04-05') SET FILEFORMAT PARQUET", tableName));
+    validWriteIds  = msClient.getValidWriteIds("default." + tableName).toString();
+    Assert.assertEquals("default.alter_table:6:9223372036854775807::", validWriteIds);
+
+    runStatementOnDriver(String.format("ALTER TABLE %s PARTITION (ds='2013-04-05') COMPACT 'minor'", tableName));
```
Comment (pvary) [inline ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands.java:470]: Why do we want to increase writeId for compaction initialization?

Comment (kishendas) [reply ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands.java:470]: That test case actually ensures we don't increase the write ID during compaction !

Comment (pvary) [reply ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands.java:470]: I see now 😄 
Maybe a comment?

Comment (kishendas) [reply ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands.java:472]: Done

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/owner/AlterTableSetOwnerAnalyzer.java [hunk 2]
```diff
@@ -26,7 +26,11 @@
 import org.apache.hadoop.hive.ql.ddl.DDLSemanticAnalyzerFactory.DDLType;
 import org.apache.hadoop.hive.ql.ddl.privilege.PrincipalDesc;
 import org.apache.hadoop.hive.ql.ddl.table.AbstractAlterTableAnalyzer;
+import org.apache.hadoop.hive.ql.ddl.table.partition.add.AlterTableAddPartitionDesc;
+import org.apache.hadoop.hive.ql.exec.Task;
 import org.apache.hadoop.hive.ql.exec.TaskFactory;
+import org.apache.hadoop.hive.ql.io.AcidUtils;
+import org.apache.hadoop.hive.ql.metadata.Table;
```
Comment (hsnusonic) [inline ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/owner/AlterTableSetOwnerAnalyzer.java:33]: Do we need all these import?

Comment (kishendas) [reply ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/owner/AlterTableSetOwnerAnalyzer.java]: Fixed it.

**Comment by kishendas** [comment]
@pvary @hsnusonic Please take a look now. Addressed all the comments and got a green run.

**Review by hsnusonic — APPROVED** [review]
LGTM

**Review by pvary — APPROVED** [review]

**Pull request merged**